tests/test-behat-tags.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+use WP_CLI\Utils;
+
+class BehatTagsTest extends PHPUnit_Framework_TestCase {
+
+	var $temp_dir;
+
+	function setUp() {
+		parent::setUp();
+
+		$this->temp_dir = Utils\get_temp_dir() . uniqid( 'wp-cli-test-behat-tags-', true );
+		mkdir( $this->temp_dir );
+		mkdir( $this->temp_dir . '/features' );
+	}
+
+	function tearDown() {
+
+		if ( $this->temp_dir && file_exists( $this->temp_dir ) ) {
+			foreach ( glob( $this->temp_dir . '/features/*' ) as $feature_file ) {
+				unlink( $feature_file );
+			}
+			rmdir( $this->temp_dir . '/features' );
+			rmdir( $this->temp_dir );
+		}
+
+		parent::tearDown();
+	}
+
+	/**
+	 * @dataProvider data_behat_tags_wp_version
+	 */
+	function test_behat_tags_wp_version( $env, $expected ) {
+		$env_wp_version = getenv( 'WP_VERSION' );
+		$env_github_token = getenv( 'GITHUB_TOKEN' );
+
+		putenv( 'WP_VERSION' );
+		putenv( 'GITHUB_TOKEN' );
+
+		$behat_tags = dirname( __DIR__ ) . '/ci/behat-tags.php';
+
+		$contents = '@require-wp-4.6 @require-wp-4.8 @require-wp-4.9 @less-than-wp-4.6 @less-than-wp-4.8 @less-than-wp-4.9';
+		file_put_contents( $this->temp_dir . '/features/wp_version.feature', $contents );
+
+		$output = exec( "cd {$this->temp_dir}; $env php $behat_tags" );
+		$this->assertSame( $expected, $output );
+
+		putenv( false === $env_wp_version ? 'WP_VERSION' : "WP_VERSION=$env_wp_version" );
+		putenv( false === $env_github_token ? 'GITHUB_TOKEN' : "GITHUB_TOKEN=$env_github_token" );
+	}
+
+	function data_behat_tags_wp_version() {
+		return array(
+			array( 'WP_VERSION=4.5', '--tags=~@require-wp-4.6&&~@require-wp-4.8&&~@require-wp-4.9&&~@github-api&&~@broken' ),
+			array( 'WP_VERSION=4.6', '--tags=~@require-wp-4.8&&~@require-wp-4.9&&~@github-api&&~@broken' ),
```

Comment:
Shouldn't this produce `~@less-than-wp-4.6` as well, as 4.6 is not less than 4.6 ?

Reply:
Ha, very good point! This means that the `@less-than-php` tags are also wrong.

Reply:
Changed `@less-than-wp` to do the right thing, and also changed `@less-than-php` as well for consistency.

Although the latter is technically a BC break it doesn't cause us issues as it's just compared against (at most) the major/minor version, and unlikely to cause others issues either. Removed unnecessary use in `features/cli.feature` also anyway.